### PR TITLE
Rename and replace 'client' in classes and comments

### PR DESCRIPTION
**Pull request by the-other-tim-brown**
## *Important Read*
- *Please ensure the GitHub issue is mentioned at the beginning of the PR*

## What is the purpose of the pull request

Addresses #406 
Renames the SourceClient, TargetClient, and OneTableClient to ConversionSource, ConversionTarget, and ConversionController.

## Brief change log

- Changed `client` package to `conversion`
- `SourceClient` and similar classes are now `ConversionSource` or a name derived from that
- `TargetClient` and similar classes are now `ConversionTarget` or a name derived from that
- `OneTableClient` is `ConversionController`
- `RunSync` configs were updated to remove references to `client`
## Verify this pull request

This pull request is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

**Comment by the-other-tim-brown** [comment]
Raising as a draft. I need to confirm the demo still works after this.

**Comment by the-other-tim-brown** [comment]
cc @lmccay

File: api/src/main/java/org/apache/xtable/spi/sync/TableFormatSync.java [hunk 1]
```diff
@@ -91,13 +91,13 @@ public Map<String, SyncResult> syncSnapshot(
    * @return the results of trying to sync each change
    */
   public Map<String, List<SyncResult>> syncChanges(
-      Map<TargetClient, TableSyncMetadata> targetClientWithMetadata,
+      Map<ConversionTarget, TableSyncMetadata> targetClientWithMetadata,
       IncrementalTableChanges changes) {
     Map<String, List<SyncResult>> results = new HashMap<>();
-    Set<TargetClient> clientsWithFailures = new HashSet<>();
+    Set<ConversionTarget> clientsWithFailures = new HashSet<>();
```
Comment (ashvina) [inline api/src/main/java/org/apache/xtable/spi/sync/TableFormatSync.java:97]: nit: rename the variable along with the type.

File: api/src/main/java/org/apache/xtable/spi/sync/TableFormatSync.java [hunk 2]
```diff
@@ -91,13 +91,13 @@ public Map<String, SyncResult> syncSnapshot(
    * @return the results of trying to sync each change
    */
   public Map<String, List<SyncResult>> syncChanges(
-      Map<TargetClient, TableSyncMetadata> targetClientWithMetadata,
+      Map<ConversionTarget, TableSyncMetadata> targetClientWithMetadata,
```
Comment (ashvina) [inline api/src/main/java/org/apache/xtable/spi/sync/TableFormatSync.java:94]: nit: rename the variable along with the type.

File: api/src/main/java/org/apache/xtable/spi/sync/TableFormatSync.java [hunk 3]
```diff
@@ -91,13 +91,13 @@ public Map<String, SyncResult> syncSnapshot(
...
       IncrementalTableChanges changes) {
     Map<String, List<SyncResult>> results = new HashMap<>();
-    Set<TargetClient> clientsWithFailures = new HashSet<>();
+    Set<ConversionTarget> clientsWithFailures = new HashSet<>();
     while (changes.getTableChanges().hasNext()) {
       TableChange change = changes.getTableChanges().next();
-      Collection<TargetClient> clientsToSync =
+      Collection<ConversionTarget> clientsToSync =
```
Comment (ashvina) [inline api/src/main/java/org/apache/xtable/spi/sync/TableFormatSync.java:100]: nit: rename the variable along with the type.

File: api/src/main/java/org/apache/xtable/spi/sync/TableFormatSync.java [hunk 4]
```diff
@@ -106,17 +106,17 @@ public Map<String, List<SyncResult>> syncChanges(
                   })
               .map(Map.Entry::getKey)
               .collect(Collectors.toList());
-      for (TargetClient targetClient : clientsToSync) {
-        if (clientsWithFailures.contains(targetClient)) {
+      for (ConversionTarget conversionTarget : clientsToSync) {
```
Comment (ashvina) [inline api/src/main/java/org/apache/xtable/spi/sync/TableFormatSync.java:109]: nit: rename clientsToSync along with the type.

File: api/src/main/java/org/apache/xtable/spi/sync/TableFormatSync.java [hunk 5]
```diff
@@ -144,7 +144,7 @@ private static boolean isChangeApplicableForLastSyncMetadata(
   }
 
   private SyncResult getSyncResult(
-      TargetClient client,
+      ConversionTarget client,
```
Comment (ashvina) [inline api/src/main/java/org/apache/xtable/spi/sync/TableFormatSync.java:147]: nit: rename client along with the type.

File: api/src/test/java/org/apache/xtable/spi/sync/TestTableFormatSync.java [hunk 6]
```diff
@@ -371,7 +373,7 @@ private DataFilesDiff getFilesDiff(int id) {
   }
 
   private void verifyBaseClientCalls(
-      TargetClient mockClient,
+      ConversionTarget mockClient,
```
Comment (ashvina) [inline api/src/test/java/org/apache/xtable/spi/sync/TestTableFormatSync.java:376]: rename mockclient to mockTarget?

File: core/src/main/java/org/apache/xtable/delta/DeltaConversionSource.java [hunk 7]
```diff
@@ -109,7 +109,7 @@ public InternalSnapshot getCurrentSnapshot() {
   @Override
   public TableChange getTableChangeForCommit(Long versionNumber) {
     InternalTable tableAtVersion = tableExtractor.table(deltaLog, tableName, versionNumber);
-    // Client to call getCommitsBacklog and call this method.
+    // Source to call getCommitsBacklog and call this method.
```
Comment (ashvina) [inline core/src/main/java/org/apache/xtable/delta/DeltaConversionSource.java:112]: Unrelated to this PR: this comment is not clear to me. Do you think more clarity is needed here?

Comment (the-other-tim-brown) [reply core/src/main/java/org/apache/xtable/delta/DeltaConversionSource.java]: I don't understand this comment, may be leftover from some dev work so I am just removing it.

**Review by ashvina — APPROVED** [review]
LGTM 👍 

Thanks for the contribution.

**Comment by the-other-tim-brown** [comment]
I have confirmed that the demo is still working after these updates

**Comment by the-other-tim-brown** [comment]
Tested the manual conversion flow documented in the README as well to confirm all the values are properly renamed in the yamls, I will merge this now.

**Pull request merged**